src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -370,6 +371,40 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
                     }
                 }
             }
+            else if (AdvSimd.IsSupported)
+            {
+                if (offset < (nuint)(uint)length)
```

Comment:
Is this to enforce zero rather than sign-extension by the JIT? Now that we have `nuint`, would it be worthwhile to do the cast to `nuint` once?

Reply:
I honestly don't know or didn't spend time debugging the `nunit` portion. I tried to use same code as used in `Sse2`/`Avx`. May be we can have a separate exercise to remove unnecessary casts through out the code.